### PR TITLE
CLOUDP-284014: Check soa migration in external path

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

_Jira ticket:_ CLOUDP-284014

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works

### Changes to Spectral
- [ ] I have read the [README](../tools/spectral/README.md) file for Spectral Updates

### CODE REVIEW

File: tools/cli/internal/openapi/oasdiff.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ func (o OasDiff) mergePaths() error {
 		if originalPathData := basePaths.Value(path); originalPathData == nil {
 			basePaths.Set(path, removeExternalRefs(externalPathData))
 		} else {
-			if err := o.handlePathConflict(originalPathData, path); err != nil {
+			if err := o.handlePathConflict(externalPathData, path); err != nil {
```
Comment (blva) [inline tools/cli/internal/openapi/oasdiff.go:79]: main change: we were getting the originalPathData (baseData, which is mms, but we expect annotations to be in SOA)

Comment (andreaangiolillo) [reply tools/cli/internal/openapi/oasdiff.go:79]: What does it mean "_expect annotations to be in SOA_"? Do you mean the file path of the spec was MMS but we need the service spec instead?

Comment (blva) [reply tools/cli/internal/openapi/oasdiff.go:79]: expect annotations to be in Standalone service, correct. We were adding the annotation to MMS, but it should live in the standalone service once the copy endpoint is created

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**